### ISSUE TITLE
Allow suitably credentialed users (developers?) to view the names of existing environment variables

### ISSUE THREAD

**Issue by tobybellwood** [uselagoon/lagoon#3404]
Currently, the RBAC matrix denies permission to "view" certain environment variables (project-scope, production env).

Ideally, non-maintainers who need to know these variables exist should be able to do so easily. Should we create a new permission - View Environment Variable Name (or similar) and grant that to all developers under the current RBAC.

This would allow them to at least know they exist, even if they aren't sure of the value.

**Comment by shreddedbacon** [comment]
Somewhat related issue #3001 with cascading/consolidating envvars to show to users

**Comment by shreddedbacon** [comment]
Well project scope variables are available to all environments of a project, so it doesn't really make sense that they're hidden from developer role.

Considering the current matrix of access
```
getEnvVarsByProjectId | env_var | project:view | projectID | Yes | Yes | Yes | Yes | No | No | No |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
getEnvVarsByEnvironmentId | env_var | environment:view:development | projectID | Yes | Yes | Yes | Yes | Yes | No | No |  
getEnvVarsByEnvironmentId | env_var | environment:view:production | projectID | Yes | Yes | Yes | Yes | No | No | No
```

**Comment by shreddedbacon** [comment]
Considering the current access matrix, there are issues with presenting some of this information in the UI already, if a reporter/guest would try to view a page that contains any variable resolver, it will error out entirely, this would be a poor user experience IMO, unless the error is extremely clear and presented in a way to the user that makes it absolutely clear they do not have permission and that asking why they don't have permission is useless. Since the main reason this issue exists is because of wanting to display this information in the UI, the UI needs to be heavily considered in how it handles these errors if we make a new tab available to view something.

Taking the above into consideration, there may be some benefit in changing the permission to be that the existing RBAC is for viewing the values of variables only. This way, if you make a request with name and scope only, the data would be returned if you have permission to `view environment/project`, this would allow all roles that currently have permission to view project/environment for resolving pages in the UI. Users would still get a permission denied error if they requested the value of these variables though.

It would be a bit of a shake up on the resolver though, you'd need some way to handle an error if the user requested the value field to keep the RBAC policy valid. The UI would need a way to have a user "reveal" it by way of a button or other method (we would have to create a new query resolver that could perform this value retrieval action, we don't have a specific get variable by name or id query at this time, only all or nothing)

I think also the benefits of this would be that everyone that has access to the project/environment now have full visibility of variable names/scopes being provided to an environment, but not necessarily knowing the values, so troubleshooting for non-developers can be partially done by checking if there is even a env var set. Those that do need to know the value, can request this information if they need it using the CLI or the "reveal" button in the UI.

Name | Resource | Scope | Attributes | Platform-Wide Admin | Platform-Wide Owner | Owner | Maintainer | Developer | Reporter | Guest | Self
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
getEnvVarsByProjectId | env_var | project:view | projectID | Yes | Yes | Yes | Yes | No | No | No |  
getEnvVarsByEnvironmentId | env_var | environment:view:development | projectID | Yes | Yes | Yes | Yes | Yes | No | No |  
getEnvVarsByEnvironmentId | env_var | environment:view:production | projectID | Yes | Yes | Yes | Yes | No | No | No
getEnvironmentsByProjectId | environment | view | projectID | Yes | Yes | Yes | Yes | Yes | Yes | Yes
getProjectByEnvironmentId | project | view | projectID | Yes | Yes | Yes | Yes | Yes | Yes | Yes

TLDR
1) https://github.com/uselagoon/lagoon/blob/main/services/api/src/resources/env-variables/resolvers.ts#L17 add a check that can possibly inspect the query fields for `value` in the request. If value is present/requested perform the `hasPermission` check and error like it normally would if the user hasn't got access, otherwise respond with the name and scopes if the user has already confirmed view access to the project/environment as existing RBAC allows this down to `guest`

2) create a new query resolver that can `get envvar by name/id in project or environment` that could be used by a reveal value button in the UI so that the above hidden value using RBAC can be performed in the UI by not retrieving the value field

**Comment by shreddedbacon** [comment]
After some discussion, we should add new scopes for `project:viewValue`, `environment:viewValue:development`, and `environment:viewValue:production`. And then update the existing `view` only to cover roles down to guest.

Name | Resource | Scope | Attributes | Platform-Wide Admin | Platform-Wide Owner | Owner | Maintainer | Developer | Reporter | Guest | Self
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
getEnvVarsByProjectId | env_var | project:viewValue | projectID | Yes | Yes | Yes | Yes | No | No | No |  
getEnvVarsByEnvironmentId | env_var | environment:viewValue:development | projectID | Yes | Yes | Yes | Yes | Yes | No | No |  
getEnvVarsByEnvironmentId | env_var | environment:viewValue:production | projectID | Yes | Yes | Yes | Yes | No | No | No
getEnvVarsByProjectId | env_var | project:view | projectID | Yes | Yes | Yes | Yes | Yes | Yes | Yes |  
getEnvVarsByEnvironmentId | env_var | environment:view:development | projectID | Yes | Yes | Yes | Yes | Yes | Yes | Yes |  
getEnvVarsByEnvironmentId | env_var | environment:view:production | projectID | Yes | Yes | Yes | Yes | Yes | Yes | Yes
getEnvironmentsByProjectId | environment | view | projectID | Yes | Yes | Yes | Yes | Yes | Yes | Yes
getProjectByEnvironmentId | project | view | projectID | Yes | Yes | Yes | Yes | Yes | Yes | Yes
